### PR TITLE
[added classnames to clickable modal components]

### PR DESCRIPTION
**Pull request by graydonpleasants**
added a proposal for passing classnames into any clickable elements in modals (for tracking, testing stability etc)

### CODE REVIEW

**Comment by MagdalenaHenke** [comment]
@Jinsoo94 @tariqporter - I'm ok with this one going through - allowing to pass in extra classes for blx-overlay and the modal close button. 
If we do that for modals, maybe we should also do that for Alerts, though.

Since this is the same context of your modals PR, I'd wanna get you into the conversation cause conflicts will happen. @Jinsoo94 you ok with this PR?

File: packages/blocks-react/components/modals/BaseModal.jsx [hunk 1]
```diff
@@ -1,30 +1,33 @@
 const React = require('react');
 const PropTypes = require('prop-types');
 const CloseIcon = require('../icons/CloseIcon.jsx');
+const classnames = require('classnames');
 
 
-const BaseModal = (props) => {
-  if (props.isHidden) return null;
+const BaseModal = ({ className, closeClassName, overlayClassName, style, isHidden, onClose, children }) => {
```
[the remark this answers is not in the document]
Comment (Jinsoo94) [reply packages/blocks-react/components/modals/BaseModal.jsx]: ^

[the remark this answers is not in the document]
Comment (MagdalenaHenke) [reply packages/blocks-react/components/modals/BaseModal.jsx]: But I am rooting for consistency - I wouldn't want us to change it _just for this component_ on this PR. What if on this PR, it stays classNames, and in a second sweep (maybe on our side), we update _all_ components to use `classes` so we keep consistency across the code base on each release?

[the remark this answers is not in the document]
Comment (tariqporter) [reply packages/blocks-react/components/modals/BaseModal.jsx]: Yeh that's fine! The component should accept `className` also though, so I think the only thing that should change here is joining those 2 props into one.

[the remark this answers is not in the document]
Comment (MagdalenaHenke) [reply packages/blocks-react/components/modals/BaseModal.jsx]: @graydonpleasants 👆 that work for you?

**Comment by MagdalenaHenke** [comment]
I'm sorry about the conflicts now.

**Comment by graydonpleasants** [comment]
Updated the Pr off the new BaseModal, I get dev:docs problems locally though, in components I haven't touched

**Comment by Jinsoo94** [comment]
how did you change `package-lock.json`?

File: packages/blocks-react/components/modals/BaseModal.jsx [hunk 2]
```diff
@@ -4,18 +4,18 @@ const CloseIcon = require('../icons/CloseIcon.jsx');
...
-        className="blx-modal-overlay"
-        onClick={props.onClose}
+        className={classnames('blx-modal-overlay', props.overlayClassName)}
+        onClick={onClose}
       />
       <div className={classnames('blx-modal', props.className)} style={props.style}>
         {
           props.isClosable && (
-            <button className="blx-modal-close" aria-label="close" onClick={props.onClose}>
+            <button className={lassnames('blx-modal-close', props.closeClassName)} aria-label="close" onClick={props.onClose}>
```
Comment (tariqporter) [inline packages/blocks-react/components/modals/BaseModal.jsx]: `classNames`?

**Comment by MagdalenaHenke** [comment]
I see package.lock changes in here now - I don't think those were intentional.

**Pull request closed without merging**